src/legacy.rs

```diff
@@ -0,0 +1,194 @@
+    ) -> Result<AesGcmTestPayload> {
+        let cryptographer = crate::crypto::holder::get_cryptographer();
+        let private_key = hex::decode(private_key).unwrap();
+        let public_key = hex::decode(public_key).unwrap();
+        let ec_key = EcKeyComponents::new(private_key, public_key);
+        let local_key_pair = cryptographer.import_key_pair(&ec_key)?;
+        let remote_pub_key = hex::decode(remote_pub_key).unwrap();
+        let remote_pub_key = cryptographer.import_public_key(&remote_pub_key).unwrap();
+        let auth_secret = hex::decode(auth_secret).unwrap();
+        let salt = hex::decode(salt).unwrap();
+        let plaintext = plaintext.as_bytes();
+        let params = WebPushParams::new(rs, pad_length, salt);
+        let encrypted_block = AesGcmEceWebPush::encrypt_with_keys(
+            &*local_key_pair,
+            &*remote_pub_key,
+            &auth_secret,
+            &plaintext,
+            params,
+        )?;
+        Ok(AesGcmTestPayload {
+            dh: hex::encode(encrypted_block.dh),
+            salt: hex::encode(encrypted_block.salt),
+            rs: encrypted_block.rs,
+            ciphertext: hex::encode(encrypted_block.ciphertext),
+        })
+    }
+
+    fn try_decrypt(
+        private_key: &str,
+        public_key: &str,
+        auth_secret: &str,
+        payload: &AesGcmTestPayload,
+    ) -> Result<String> {
+        let private_key = hex::decode(private_key).unwrap();
+        let public_key = hex::decode(public_key).unwrap();
+        let ec_key = EcKeyComponents::new(private_key, public_key);
+        let plaintext = decrypt_aesgcm(
+            &ec_key,
+            &hex::decode(auth_secret).unwrap(),
+            &AesGcmEncryptedBlock::new(
+                &hex::decode(&payload.dh).unwrap(),
+                &hex::decode(&payload.salt).unwrap(),
+                payload.rs,
+                hex::decode(&payload.ciphertext).unwrap(),
+            )?,
+        )?;
+        Ok(String::from_utf8(plaintext).unwrap())
+    }
+
+    #[test]
+    fn test_e2e() {
+        let (local_key, remote_key) = crate::generate_keys().unwrap();
+        let plaintext = b"There was a green mouse, running in the grass";
+        let mut auth_secret = vec![0u8; 16];
+        let cryptographer = crate::crypto::holder::get_cryptographer();
+        cryptographer.random_bytes(&mut auth_secret).unwrap();
+        let remote_public = cryptographer
+            .import_public_key(&remote_key.pub_as_raw().unwrap())
+            .unwrap();
+        let params = WebPushParams::default();
+        let encrypted_block = AesGcmEceWebPush::encrypt_with_keys(
+            &*local_key,
+            &*remote_public,
+            &auth_secret,
+            plaintext,
+            params,
+        )
+        .unwrap();
+        let decrypted =
+            AesGcmEceWebPush::decrypt(&*remote_key, &auth_secret, &encrypted_block).unwrap();
+        assert_eq!(decrypted, plaintext.to_vec());
+    }
```

Comment:
This test is very similar to the one in [the main `lib.rs`](https://github.com/mozilla/rust-ece/blob/64af2968ebaab43b5a2206795ad64de3153e585f/src/lib.rs#L410), except that your one uses an explicit salt. What do you think about replacing the one linked above with your version here rather than having two nearly-identical tests?